pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -652,15 +652,11 @@ def _validate_monitor_key(self, trainer: 'pl.Trainer') -> None:
 
         # validate metric
         if self.monitor is not None and not self._is_valid_monitor_key(metrics):
-            m = (
+            raise MisconfigurationException(
```

Comment:
So would this raise an exception even if the val_loop hasn't run ?

Reply:
_validate_monitor_key would be called whenever the checkpoint is going to be saved. this could happen after training batches too, depending on the callback configuration. checking if the val loop has run, and if ithasn't, raising a warning vs if it has, raising an exception, silently hides this case where the monitor metric is misconfigured. 

i think its fair to enforce that if a metric key is provided, it must be present at the time we're saving a checkpoint